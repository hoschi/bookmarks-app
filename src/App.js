import * as R from 'ramda'
import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import Grid from '@material-ui/core/Grid'
import BookmarksList from './BookmarksList'

const styles = R.always({
    root: {
        margin: 16,
    },
    heading: {
        textAlign: 'center',
    },
})

function App({ classes }) {
    return (
        <div className={classes.root}>
            <Grid container spacing={16}>
                <Grid item xs={12}>
                    <Typography component="h1" variant="display3" className={classes.heading}>
                        Bookmarks
                    </Typography>
                </Grid>
                <Grid item xs={12}>
                    <BookmarksList />
                </Grid>
            </Grid>
        </div>
    )
}

App.propTypes = {
    classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(App)
