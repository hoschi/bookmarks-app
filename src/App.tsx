import { Box, Container, Typography } from '@mui/material'
import { orange } from '@mui/material/colors'
import styled from '@emotion/styled'

import { MyButton } from './MyButton'

const StyledDiv = styled.div`
    padding: 32px;
    background-color: ${orange.A400};
    font-size: 24px;
    border-radius: 4px;
    color: black;
    font-weight: bold;
    &:hover {
        color: white;
    }
`

export const App: React.FunctionComponent = () => {
    return (
        <Container maxWidth="sm">
            <Box sx={{ my: 4 }}>
                <Typography variant="h4" component="h1" sx={{ mb: 2 }}>
                    Tracking Reporter
                </Typography>
                <MyButton />
                <StyledDiv>blubs</StyledDiv>
            </Box>
        </Container>
    )
}
