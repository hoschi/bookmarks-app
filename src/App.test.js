import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

test('render', function() {
    const actual = shallow(<App />).dive()
    expect(actual).toMatchSnapshot()
})
