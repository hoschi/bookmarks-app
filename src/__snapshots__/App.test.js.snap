// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="App-root-1"
>
  <WithStyles(Grid)
    container={true}
    justify="center"
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        className="App-heading-2"
        component="h1"
        variant="display3"
      >
        Bookmarks
      </WithStyles(Typography)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="App-content-3"
      item={true}
      xs={12}
    >
      <BookmarkAdd />
      <BookmarksList />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
